addon_service/tests/_helpers.py

```diff
@@ -159,6 +173,70 @@ async def _route_post(self, url, *args, **kwargs):
             raise RuntimeError(f"Received unrecognized endpoint {url}")
 
 
+@dataclasses.dataclass
+class MockServiceProvider:
```

Comment:
naming nit: since the code is specific to oauth1, how about `MockOauth1ServiceProvider` or `MockOauth1Server`?